controllers/containerjfr_controller.go

```diff
@@ -382,6 +429,50 @@ func (r *ContainerJFRReconciler) createRouteForService(controller *rhjmcv1beta1.
 	}, nil
 }
 
+func (r *ContainerJFRReconciler) createIngressForService(controller *rhjmcv1beta1.ContainerJFR, svc *corev1.Service, exposePort corev1.ServicePort,
+	tlsConfig *openshiftv1.TLSConfig, networkConfig *rhjmcv1beta1.NetworkConfiguration) (*url.URL, error) {
+	logger := r.Log.WithValues("Request.Namespace", svc.Namespace, "Name", svc.Name, "Kind", fmt.Sprintf("%T", &netv1.Ingress{}))
+
+	ingress := &netv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        svc.Name,
+			Namespace:   svc.Namespace,
+			Annotations: networkConfig.Annotations,
+			Labels:      networkConfig.Labels,
+		},
+		Spec: *networkConfig.IngressSpec,
+	}
+	if err := controllerutil.SetControllerReference(controller, ingress, r.Scheme); err != nil {
+		return nil, err
+	}
+
+	found := &netv1.Ingress{}
+	err := r.Client.Get(context.Background(), types.NamespacedName{Name: svc.Name, Namespace: svc.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Not found")
+		if err := r.Client.Create(context.Background(), ingress); err != nil {
+			logger.Error(err, "Could not be created")
+			return nil, err
+		}
+		logger.Info("Created")
+		found = ingress
+	} else if err != nil {
+		logger.Error(err, "Could not be read")
+		return nil, err
+	}
+
+	logger.Info("Ingress created", "Service.Status", fmt.Sprintf("%#v", found.Status))
+	host := ""
+	if networkConfig.IngressSpec.Rules != nil && networkConfig.IngressSpec.Rules[0].Host != "" {
+		host = networkConfig.IngressSpec.Rules[0].Host
+	}
+
+	return &url.URL{
+		Scheme: getProtocol(tlsConfig),
```

Comment:
This is one thing that I am not sure about. Is it relevant to get the protocol from the tlsConfig at all, or should it just be hardocoded? I was looking if it can be specified in the ingressSpecs, bit in the ingressRule it says only HTTP is supported so would that just be hardcoded into the Scheme field?

Reply:
I think the protocol is determined by the ingress's spec.tls property. It's probably good enough to use `https` if spec.tls is non-empty.